frontend/app/controllers/Promotions.scala

```diff
@@ -102,7 +104,9 @@ object Promotions extends Controller {
       for {
         promotion <- promoService.findPromotion(promoCode) \/> notFound //if we can't find the promotion fail out of the for comprehension with a notFound
         _ <- failWhen(promoCodeStr.toUpperCase != promoCodeStr,redirectToUpperCase)
+        _ <- failWhen(promotion.starts.isAfterNow,notFound)
         _ <- failWhen(promotion.expires.isBeforeNow,notFound)
+        _ <- failWhen(promotion.whenTracking.isDefined,redirectToHomePage)
```

Comment:
Am I right that this uses `failWhen` for a valid behaviour? If so that feels a bit odd.